share/wake/lib/core/list.wake

```diff
@@ -23,46 +23,61 @@ package wake
 #   - the comma operator to prepend elements to a List
 #
 # Because the comma operator is right associative, a List parses like this:
+#   ```
 #   0, 1, 2, Nil = (0, (1, (2, Nil)))
+#   ```
 #
 # Since List has a type parameter, any type can be legally stored in a list
+#   ```
 #   0, 1, 2, Nil         # ok! This is a `List Integer`
 #   "0", "1", "2", Nil   # ok! This is a `List String`
+#   ```
 #
-# However, mixing types it NOT legal:
+# However, mixing types is *not* legal:
+#   ```
 #   0, "1", Nil          # bad! This will trigger a type-check failure.
+#   ```
 export data List a =
     Nil
     (head: a), (tail: List a)
 
-# empty: report if the list contains no elements.
+# Report if the list contains no elements.
 # If you find yourself using the function, consider using match instead.
 #
+# Examples:
+#   ```
```

Comment:
I notice you're indenting the triple-backtics to line up with the code inside. In wake itself, you would indent the contents of the opening escape, for example:

```
def foo =
  """
    This is a multi-line string.
    Hello world.
  """
```

Or:
```
def foo = (
  def bar = foo + foo
  foo
  + bar
  + bar
)
```

... to be consistent with code, should we follow this policy also within the comments?

Reply:
So, I guess I am advocating for either:
```
# Examples:
# ```
#   foo x = True
# ```

or

# Examples:
#   ```
#     foo x = True
#   ```
```

Reply:
Sure. I think of those two options I prefer the first (quotes unintended but code remains indented). I'll switch that over